src/main/groovy/com/blackducksoftware/integration/hub/artifactory/scan/blackDuckScanForHub.groovy

```diff
@@ -476,23 +476,28 @@ private void writeScanProperties(RepoPath repoPath, ProjectVersionView projectVe
-            projectVersionUrl = updateUrlPropertyToCurrentHubServer(projectVersionUrl)
-            repositories.setProperty(it, BLACK_DUCK_PROJECT_VERSION_URL_PROPERTY_NAME, projectVersionUrl)
-            ProjectVersionView projectVersionView = hubResponseService.getItem(projectVersionUrl, ProjectVersionView.class)
-            try{
-                String policyStatusUrl = metaService.getFirstLink(projectVersionView, MetaService.POLICY_STATUS_LINK)
-                log.info("Looking up policy status: ${policyStatusUrl}")
-                VersionBomPolicyStatusView versionBomPolicyStatusView = hubResponseService.getItem(policyStatusUrl, VersionBomPolicyStatusView.class)
-                log.info("policy status json: ${versionBomPolicyStatusView.json}")
-                PolicyStatusDescription policyStatusDescription = new PolicyStatusDescription(versionBomPolicyStatusView)
-                repositories.setProperty(it, BLACK_DUCK_POLICY_STATUS_PROPERTY_NAME, policyStatusDescription.policyStatusMessage)
-                repositories.setProperty(it, BLACK_DUCK_OVERALL_POLICY_STATUS_PROPERTY_NAME, versionBomPolicyStatusView.overallStatus.toString())
-                log.info("Added policy status to ${it.name}")
-            }catch(HubIntegrationException e){
-                problemRetrievingPolicyStatus = true
+        try {
+            String projectVersionUrl = repositories.getProperty(it, BLACK_DUCK_PROJECT_VERSION_URL_PROPERTY_NAME)
+            if (StringUtils.isNotBlank(projectVersionUrl)) {
+                projectVersionUrl = updateUrlPropertyToCurrentHubServer(projectVersionUrl)
+                repositories.setProperty(it, BLACK_DUCK_PROJECT_VERSION_URL_PROPERTY_NAME, projectVersionUrl)
+                ProjectVersionView projectVersionView = hubResponseService.getItem(projectVersionUrl, ProjectVersionView.class)
+                try{
+                    String policyStatusUrl = metaService.getFirstLink(projectVersionView, MetaService.POLICY_STATUS_LINK)
+                    log.info("Looking up policy status: ${policyStatusUrl}")
+                    VersionBomPolicyStatusView versionBomPolicyStatusView = hubResponseService.getItem(policyStatusUrl, VersionBomPolicyStatusView.class)
+                    log.info("policy status json: ${versionBomPolicyStatusView.json}")
+                    PolicyStatusDescription policyStatusDescription = new PolicyStatusDescription(versionBomPolicyStatusView)
+                    repositories.setProperty(it, BLACK_DUCK_POLICY_STATUS_PROPERTY_NAME, policyStatusDescription.policyStatusMessage)
+                    repositories.setProperty(it, BLACK_DUCK_OVERALL_POLICY_STATUS_PROPERTY_NAME, versionBomPolicyStatusView.overallStatus.toString())
+                    log.info("Added policy status to ${it.name}")
+                } catch(HubIntegrationException e){
+                    problemRetrievingPolicyStatus = true
+                }
             }
+        } catch(Exception e){
```

Comment:
Logic-wise, this is fine. Logging wise, this is a little murky. 

I think we want to have two catch blocks, one that catches the exact locking exception that we know about and likely requires customer remediation (if a file is locked for 2+ minutes, we can safely assume that something went wrong- hence why the exception gets thrown) so we can log a very explicit error. The error would probably be something like what you have, but perhaps a bit more specifically related to the lock.

The other should catch any generic exception (so we don't stop looping through repoPaths) and log the stack trace.

Also, minor whitespace suggestion: put a space between the ending paren and curly brace, like on line 481.